fenix/app/src/main/java/org/mozilla/fenix/settings/SecretSettingsFragment.kt

```diff
@@ -86,6 +87,18 @@ class SecretSettingsFragment : PreferenceFragmentCompat() {
             }
         }
 
+        requirePreference<SwitchPreference>(R.string.pref_key_enable_debug_drawer).apply {
+            isVisible = Config.channel.isNightlyOrDebug
+            isChecked = context.settings().enableDebugDrawer
+            onPreferenceChangeListener =
+                Preference.OnPreferenceChangeListener { _, newValue ->
+                    context.settings().enableDebugDrawer = newValue as Boolean
+                    context.components.appStore.dispatch(AppAction.UpdateDebugDrawerEnabled(newValue))
+
+                    true
```

Comment:
IMO if we are going to manage this state in a Store, then a middleware should be handling the `Settings` side-effects